Birthday-2024-Project/Scripts/Game/States/GameState.gd

```diff
@@ -3,22 +3,53 @@ extends Node2D
 
 var manager: GameManager
 
-
-func enter_state():
-	pass
-
+func next_puzzle(skipConfirm : bool = true):
+	if skipConfirm:
+		manager.myScreen.ScreenEnter.connect(SkipConfirmation)
+		manager.myScreen.ShowConfirmationPopup("Skip Level", "Do you want to skip to the next level?", "Yes", "No")
+	else:
+		manager.go_to_next_level()
+		
+func SkipConfirmation():
+	manager.myScreen.ScreenEnter.disconnect(SkipConfirmation)
+	if manager.myScreen.transitionData[ConfirmationPopupController.RESPONSE_KEY]:
+		manager.go_to_next_level()
+
+func reset_puzzle():
+	manager.myScreen.ScreenEnter.connect(ResetConfirmation)
+	manager.myScreen.ShowConfirmationPopup("Restart", "Clear pieces and restart from the beginning?", "Yes", "No")
+	
+func ResetConfirmation():
+	manager.myScreen.ScreenEnter.disconnect(ResetConfirmation)
+	if manager.myScreen.transitionData[ConfirmationPopupController.RESPONSE_KEY]:
+		_reset_puzzle()
+
+func _reset_puzzle():
+	manager.grid.ReloadLevel()
+
+func back_to_menu(askConfirm : bool = true):
+	if askConfirm:
+		manager.myScreen.ScreenEnter.connect(BackToMenuConfirmation)
+		manager.myScreen.ShowConfirmationPopup("Main Menu", "Go back to the Main Menu?", "Yes", "No")
+	else:
+		manager.go_to_main_menu()
```

Comment:
see the comment above, we should change the name of go_to_main_menu to e.g. go_to_level_select to avoid any misunderstandings

Reply:
changed